bundles/AdminBundle/Controller/Admin/DataObject/DataObjectHelperController.php

```diff
@@ -2488,4 +2488,168 @@ protected function processAvailableFieldDefinitions($fds, &$firstOne, &$commonFi
             }
         }
     }
+    
+    /**
+     * @Route("/export-json-config", methods={"GET"})
+     *
+     * @param Request $request
+     * 
+     * @return Response
+     */
+    public function exportJsonConfigAction(Request $request)
+    {
+        $importConfigId = $request->get('importConfigId');
+        
+        if($importConfigId == null || empty($importConfigId)){
+            throw new \Exception("Please provide a valid import configuration Id");
+        }
+
+        try {
+            $importConfig = ImportConfig::getById($importConfigId);
+        
+            $configData = $importConfig->getConfig();
+            $configName = $importConfig->getName();
+            
+            $response = new Response();
+
+            $response->setContent($configData);
+
+            $disposition = $response->headers->makeDisposition(
+                ResponseHeaderBag::DISPOSITION_ATTACHMENT,
+                $configName.".json"
+            );
+
+            $response->headers->set('Content-Type', 'application/json');
+            $response->headers->set('Content-Disposition', $disposition);
+
+            return $response;
+            
+        } catch (\Exception $e) {
+            throw new \Exception("Error retrieving import configuration - ".$e->getMessage());
+        }
+    }
+    
+    /**
+     * @Route("/import-json-config", methods={"POST"})
+     *
+     * @param Request $request
+     * @param ImportService $importService
+     *
+     * @return JsonResponse
+     */
+    public function importJsonConfigAction(Request $request, ImportService $importService)
+    {
+        $importConfigId = $request->get('importConfigId');
+        
+        $tmpName = $_FILES['Filedata']['tmp_name'];
+        $json = file_get_contents($tmpName);
+
+        $configData = json_decode($json, true);
+
+        $selectedGridColumns = $configData['selectedGridColumns'];
+        $resolverSettings = $configData['resolverSettings'];
+        $shareSettings = $configData['shareSettings'];
+        $dialect = json_decode(json_encode($configData['csvSettings']), false);
+        
+        $success = true;
+        $supportedFieldTypes = ['checkbox', 'country', 'date', 'datetime', 'href', 'image', 'input', 'language', 'table', 'multiselect', 'numeric', 'password', 'select', 'slider', 'textarea', 'wysiwyg', 'objects', 'multihref', 'geopoint', 'geopolygon', 'geopolyline', 'geobounds', 'link', 'user', 'email', 'gender', 'firstname', 'lastname', 'newsletterActive', 'newsletterConfirmed', 'countrymultiselect', 'objectsMetadata'];
+
+        $classId = $request->get('classId');
+        $file = PIMCORE_SYSTEM_TEMP_DIRECTORY . '/import_' . $request->get('importId');
```

Comment:
why are we dealing with csv files suddenly? my understanding is that we want to import the exported JSON configuration file?

Reply:
My idea was to produce the same JSON response of the importGetFileInfoAction function.
The difference is that the json configuration is taken from the uploaded file instead of being loaded from database